README.md

```diff
@@ -9,183 +9,71 @@ Maxwell is an HTTP client that provides a common interface over [:httpc](http://
 
-```elixir
-$ MIX_ENV=TEST iex -S mix
-iex(1)> GitHubClient.
-edit_repo_desc/4    get!/0              get!/1
-get!/2              get/0               get/1
-patch!/0            patch!/1            patch!/2
-patch/0             patch/1             user_repos/1
-iex(1)> GitHubClient.user_repos("zhongwencool")
-22:23:42.307 [info]  get https://api.github.com <<<200(3085.772ms)
-%Maxwell.Conn{method: :get, opts: [connect_timeout: 3000, recv_timeout: 20000]
-...(truncated)
-```
-
-You can also use Maxwell without defining a module:
+The same functions are also exported by the `Maxwell` module, which you can use if you do not wish to define a wrapper
+module for your API, as shown below:
 
 ```elixir
 iex(1)> alias Maxwell.Conn
-iex(2)> Conn.new("http://httpbin.org") |>
-    Conn.put_path("/drip") |> 
+iex(2)> Conn.new("http://httpbin.org/drip") |>
     Conn.put_query_string(%{numbytes: 25, duration: 1, delay: 1, code: 200}) |> 
     Maxwell.get
 {:ok,
- %Maxwell.Conn{method: :get, opts: [], path: "",
+ %Maxwell.Conn{method: :get, opts: [], path: "/drip",
   query_string: %{code: 200, delay: 1, duration: 1, numbytes: 25},
   req_body: nil, req_headers: %{}, resp_body: '*************************',
-  resp_headers: %{"access-control-allow-credentials" => {"access-control-allow-credentials",
-     "true"},
-    "access-control-allow-origin" => {"access-control-allow-origin", "*"},
-    "connection" => {"connection", "keep-alive"},
-    "content-length" => {"content-length", "25"},
-    "content-type" => {"content-type", "application/octet-stream"},
-    "date" => {"date", "Sun, 18 Dec 2016 14:32:38 GMT"},
-    "server" => {"server", "nginx"}}, state: :sent, status: 200,
-  url: "http://httpbin.org/drip"}}
+  resp_headers: %{"access-control-allow-credentials" => "true",
+    "access-control-allow-origin" => "*",
+    "connection" => "keep-alive",
+    "content-length" => "25",
+    "content-type" => "application/octet-stream",
+    "date" => "Sun, 18 Dec 2016 14:32:38 GMT",
+    "server" => "nginx"}, state: :sent, status: 200,
+  url: "http://httpbin.org"}}
 ```
 
-### Helper functions for Maxwell.Conn
-
-```elixir
-new(request_url_string)
-|> put_query_string(request_query_map)
-|> put_req_header(request_headers_map)
-|> put_option(request_opts_keyword_list)
-|> put_req_body(request_body_term)
-|> YourClient.{http_method}!
-|> get_resp_body
-```
-
-See the documentation of `Maxwell.Conn` for more information.
-
-## Responses
-
-When calling one of (non-bang versions) of the HTTP method functions on a client module or the `Maxwell` module, you
-can expect either `{:ok, Maxwell.Conn.t}` or `{:error, reason, Maxwell.Conn.t}` to be returned.
-
-When calling of the bang versions of the HTTP method functions, e.g. `get!`, you can expect `Maxwell.Conn.t` if successful,
-or a `Maxwell.Error` will be raised.
-
-## Example Client
-
-The following is a full implementation of a client showing various features of Maxwell.
-
-```elixir
```

Comment:
I'm actually planning on improving the docs around that, but I decided to hold off until this PR is merged. I think improving the adapter docs as well as having a more in depth description around streaming and files in the readme is preferable to a big example client which leaves out the details. I can make that part of the PR if you want, or after it's merged, your call.